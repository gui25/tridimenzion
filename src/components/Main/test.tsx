import React from 'react'
import { render, screen } from '@testing-library/react'
import LandingPage from './'

test('LandingPage renders correctly', () => {
  const { asFragment } = render(<LandingPage />)

  const titleElement = screen.getByText('Explore our Stunning 3D Models')
  expect(titleElement).toBeInTheDocument()

  const descriptionElement = screen.getByText('Select a model to view in 3D:')
  expect(descriptionElement).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})
