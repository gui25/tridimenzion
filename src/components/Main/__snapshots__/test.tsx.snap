// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
<DocumentFragment>
  <main
    class="bg-gradient-to-r from-indigo-900 to-blue-900 text-white min-h-screen pb-20 md:pb-0 pt-20 md:pt-0 flex flex-col items-center justify-center"
  >
    <div
      class="max-w-xl mx-auto text-center"
    >
      <h1
        class="text-3xl md:text-5xl font-bold mb-4"
      >
        Explore our Stunning 3D Models
      </h1>
      <h2
        class="text-xl md:text-3xl font-normal mb-8"
      >
        Select a model to view in 3D:
      </h2>
    </div>
    <div
      class="flex flex-col md:flex-row justify-center mt-4 gap-6 md:gap-14"
    >
      <div
        class="bg-gray-800 text-white rounded-lg overflow-hidden max-w-lg mx-auto my-8 shadow-lg hover:shadow-xl w-120 pointer-events-none"
      >
        <div
          class="w-full h-64"
        >
          <div
            class="flex h-full w-full items-center justify-center"
          >
            <svg
              class="animate-spin flex items-center justify-center w-24 h-64"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
              />
            </svg>
          </div>
          <div
            class="relative w-full h-64 overflow-hidden pointer-events-auto touch-action-none"
            style="opacity: 0;"
          >
            <div
              style="position: relative; width: 100%; height: 100%; overflow: hidden; pointer-events: auto;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <canvas
                  style="display: block;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="p-8"
        >
          <h3
            class="text-2xl font-bold mb-4"
          >
            Diamond 3D Model
          </h3>
          <p
            class="text-gray-300 text-xl mb-8"
          >
            Explore the Diamond model in 3D
          </p>
        </div>
      </div>
      <div
        class="bg-gray-800 text-white rounded-lg overflow-hidden max-w-lg mx-auto my-8 shadow-lg hover:shadow-xl w-120 pointer-events-none"
      >
        <div
          class="w-full h-64"
        >
          <div
            class="flex h-full w-full items-center justify-center"
          >
            <svg
              class="animate-spin flex items-center justify-center w-24 h-64"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
              />
            </svg>
          </div>
          <div
            class="relative w-full h-64 overflow-hidden pointer-events-auto touch-action-none"
            style="opacity: 0;"
          >
            <div
              style="position: relative; width: 100%; height: 100%; overflow: hidden; pointer-events: auto;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <canvas
                  style="display: block;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="p-8"
        >
          <h3
            class="text-2xl font-bold mb-4"
          >
            AbstractShapes 3D Model
          </h3>
          <p
            class="text-gray-300 text-xl mb-8"
          >
            Explore the AbstractShapes model in 3D
          </p>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
